scala/scala_import.bzl

```diff
@@ -49,34 +68,21 @@ def _code_jars_and_intellij_metadata_from(jars):
            source_jars = [],
        )
      )
-  return struct(code_jars = code_jars, intellij_metadata = intellij_metadata)
+  return intellij_metadata
 
 def _filter_out_non_code_jars(files):
   return [file for file in files.to_list() if not _is_source_jar(file)]
 
 def _is_source_jar(file):
   return file.basename.endswith("-sources.jar")
 
-# TODO: it seems this could be reworked to use java_common.merge
-def _collect(deps):
-  transitive_compile_jars = []
-  runtime_jars = []
-  compile_jars = []
+def _labels_to_provider(labels):
```

Comment:
labels here is overloaded. I thought it's the actual labels. Any other ideas?

Reply:
This the purpose of this method - take a list of labels and merge them to a single `JavaInfo`